src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/tests/test_storage_validators.py

```diff
@@ -173,5 +173,75 @@ def test_validate_encryption_source(self):
         self.assertEqual(properties.key_vault_uri, 'https://example.com/key_uri')
 
 
+class TestGetSourceClientValidator(unittest.TestCase):
+    def test_validate_namespace_processor(self):
+        from azure.cli.command_modules.storage._validators import get_source_file_or_blob_service_client
+
+        # source container name given, validator does not change namespace aside from ensuring source-client none
+        ns = Namespace(account_key='my_storage_key', account_name='storage_name', connection_string=None,
+                       dryrun=False, pattern='*', sas_token=None, source_account_key=None, source_account_name=None,
+                       source_client=None, source_container='container2', source_sas=None, source_share=None,
+                       source_uri=None, destination_container='container1')
+
+        get_source_file_or_blob_service_client(ns)
+        self.assertIsNone(ns.source_client)
+
+        # source given in form of uri, source_container parsed from uri, source and dest account are the same
+        ns.source_container = None
+        ns.source_uri = 'https://storage_name.blob.core.windows.net/container2'
+
+        get_source_file_or_blob_service_client(ns)
+        self.assertEqual(ns.source_container, 'container2')
+        self.assertIsNone(ns.source_client)
+
+        # source given in form of uri, source_container parsed from uri, source and dest account are different
+        ns.source_container = None
+        ns.source_uri = 'https://other_name.blob.core.windows.net/container2?some_sas_token'
+
+        get_source_file_or_blob_service_client(ns)
+        self.assertEqual(ns.source_container, 'container2')
```

Comment:
Great test and good coverage. However, it is helpful to split this test method into smaller one. There are a few benefits:
1) Test failure is more accurate when test body is small. It renders more accurate test report.
2) A large test will stop execute whenever any assertion fails, therefore, the following test logic is skipped. It then runs fewer tests when test logic are distributed among smaller tests.

Overall, a unit test should contain only three stages:
1) Arrange
2) Act
3) Assert